scripts/imgtool/main.py

```diff
@@ -254,6 +254,10 @@ def convert(self, value, param, ctx):
               type=click.Choice(['128','256']),
               help='When encrypting the image using AES, select a 128 bit or '
                    '256 bit key len.')
+@click.option('-c', '--clear', required=False, is_flag=True, default=False,
+              help='Output an encrypted image file that can be installed on '
+                   'primary slot. (clear image code with encrypt capability '
+                   'in secondary slot during swap.)')
```

Comment:
This is incomplete. You also need to define this as an input to sign and forward it to create
```
def sign(key, public_key_format, align, version, pad_sig, header_size,
         pad_header, slot_size, pad, confirm, max_sectors, overwrite_only,
         endian, encrypt_keylen, encrypt, clear, infile, outfile, dependencies,
         load_addr, hex_addr, erased_val, save_enctlv, security_counter,
         boot_record, custom_tlv, rom_fixed, max_align):
```
in line 306
and
```
    img.create(key, public_key_format, enckey, dependencies, boot_record,
               custom_tlvs, int(encrypt_keylen), clear)
```   
in line 356
are needed to have this option accepted and forwarded. (I tested the tool with those additions and then got the expected binary which was encrypted on swap on my target).

Reply:
Correct, thanks.